lib/content-services/src/lib/content-metadata/services/content-metadata.service.ts

```diff
@@ -87,13 +86,25 @@ export class ContentMetadataService {
     }
 
     setTitleToNameIfNotSet(propertyGroups: OrganisedPropertyGroup[]): OrganisedPropertyGroup[] {
+        const propertyGroupsTitles = {};
         propertyGroups.map((propertyGroup) => {
-            propertyGroup.title = propertyGroup.title || propertyGroup.name;
+            const title = propertyGroup.title;
+            const name = propertyGroup.name;
+            if (title) {
+                if (propertyGroupsTitles[title]) {
+                    propertyGroup.title = name ? `${title} (${name})` : title;
+                } else {
+                    propertyGroup.title = title;
+                }
+                propertyGroupsTitles[title] = title;
+            } else {
+                propertyGroup.title = name;
```

Comment:
I think we should also be keeping track of cases where the name property is used for a title. For e.g., there might be a situation where there are 2 aspects, where one does not have a title, rather a name field, with the same value as the title field of the other aspect. This may be defined as follows - 
```
aspect1Group = {
        name: 'Aspect:Custom',
        properties: [...]
};

aspect2Group = {
        name: 'test:aspect2',
        title: 'Aspect:Custom',
        properties: [...]
};
```
In this case, when creating the aspect panels, for the first aspect, the code will execute the outer most else case, and the name field will be used for the title (`Aspect:Custom`). For the second aspect, the code will execute the first if block. Within it, it will check if it is duplicating any of the current aspects, which it WONT find, since the data is only being tracked for the if block, not the else block. It will then simply use the title field for the header (`Aspect:Custom`), thereby duplicating the entries, and reproducing the issue

Reply:
@swapnil-verma-gl thanks for pointing that out. I've applied the suggested change and added a new unit test to cover it.

Reply:
Thanks, this looks much better now.